states/remote/state_test.go

```diff
@@ -302,6 +302,168 @@ func TestWriteStateForMigration(t *testing.T) {
 		},
 	}
 
+	testCases := []migrationTestCase{
```

Comment:
The diff implies this is the new section, however, it is the previous set of test cases with the EnableForcePush-specific code removed. The test `TestWriteStateForMigrationWithForcePushClient` is the new test.